### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

File: src/InitializerInterface.php [hunk 1]
```diff
@@ -29,8 +29,7 @@ interface InitializerInterface extends Initializer\InitializerInterface
     /**
      * Initialize
      *
-     * @param mixed $instance
      * @return mixed
      */
-    public function initialize($instance, ServiceLocatorInterface $serviceLocator);
+    public function initialize(mixed $instance, ServiceLocatorInterface $serviceLocator);
```
Comment (Ocramius) [inline src/InitializerInterface.php:34]: Can we add this to interfaces? Seems unsafe 🤔

File: src/Tool/FactoryCreator.php [hunk 2]
```diff
@@ -100,22 +100,19 @@ private function getConstructorParameters($className)
...
             $constructorParameters,
-            function (ReflectionParameter $argument): bool {
+            static function (ReflectionParameter $argument): bool {
                 if ($argument->isOptional()) {
                     return false;
                 }
-
                 $type  = $argument->getType();
                 $class = $type instanceof ReflectionNamedType && ! $type->isBuiltin() ? $type->getName() : null;
-
```
Comment (Ocramius) [inline src/Tool/FactoryCreator.php]: Careful with spacing removals

**Review by Ocramius — CHANGES_REQUESTED** [review]
Unsure if CI failures are caused by this patch or not 🤔

File: src/InitializerInterface.php (hunk 1 above)
Comment (samsonasik) [reply src/InitializerInterface.php:34]: It already in `PluginManagerInterface` as well, see 

https://github.com/laminas/laminas-servicemanager/pull/157/files#diff-891bff41f1394c816120c19c5f4f7cfc98fa92047c8efae9b4ccd9f51ea83eae

File: src/Tool/FactoryCreator.php (hunk 2 above)
Comment (samsonasik) [reply src/Tool/FactoryCreator.php]: Ok, rolled back spacing on https://github.com/laminas/laminas-servicemanager/pull/157/commits/455ca05a4a71c5d966a7021cf15f807c0386713b

**Comment by samsonasik** [comment]
All green 🎉

File: src/AbstractPluginManager.php [hunk 3]
```diff
@@ -177,7 +177,7 @@ public function get($name, ?array $options = null)
      *
      * @psalm-assert InstanceType $instance
      */
-    public function validate($instance)
+    public function validate(mixed $instance)
```
Comment (Ocramius) [inline src/AbstractPluginManager.php:180]: Should be OK: https://3v4l.org/91RNu

File: src/ServiceManager.php [hunk 4]
```diff
@@ -562,7 +560,7 @@ private function getFactory(string $name): callable
      */
     private function createDelegatorFromName(string $name, ?array $options = null)
     {
-        $creationCallback = function () use ($name, $options) {
+        $creationCallback = function () use ($name, $options): object {
```
Comment (Ocramius) [inline src/ServiceManager.php]: This cannot be `: object` IMO: a delegator could produce anything

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/ServiceManager.php (hunk 4 above)
Comment (samsonasik) [reply src/ServiceManager.php]: Ok, rolled back https://github.com/laminas/laminas-servicemanager/pull/157/commits/0c9036e4d78bbbbe9c9c26a7c5fc05e11d77f6f8

**Comment by samsonasik** [comment]
All green 🎉

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**